src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -819,10 +826,16 @@ private void SendDocumentsBatch(DocumentsOperationContext documentsContext, Outg
             if (_log.IsInfoEnabled && _orderedReplicaItems.Count > 0)
                 _log.Info($"Finished sending replication batch. Sent {_orderedReplicaItems.Count:#,#;;0} documents and {_replicaAttachmentStreams.Count:#,#;;0} attachment streams in {sw.ElapsedMilliseconds:#,#;;0} ms. Last sent etag = {_lastEtag:#,#;;0}");
 
-            var (type, _) = _parent.HandleServerResponse();
+            var (type, reply) = _parent.HandleServerResponse(getFullResponse: true);
             if (type == ReplicationMessageReply.ReplyType.MissingAttachments)
             {
                 MissingAttachmentsInLastBatch = true;
+
+                if (_parent.MissingAttachmentsRetries.ContainsKey(reply.Exception) == false)
+                    _parent.MissingAttachmentsRetries.Add(reply.Exception, 1);
+                else
+                    _parent.MissingAttachmentsRetries[reply.Exception]++;
```

Comment:
I don't think it should be a dictionary... especially not with the exception as the key.
why not just to do a `MissingAttachmentsRetries++` and clear it on a successful response type?

Reply:
It's a dictionary because we thought it should only throw when failing more than once for the same document

Reply:
But if we are getting `MissingAttachment` we add all attachments, not only the missing one, right?

Reply:
True. Done.